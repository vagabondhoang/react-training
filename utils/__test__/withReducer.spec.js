import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import withReducer from '../withReducer';

Enzyme.configure({ adapter: new Adapter() });

describe('withReducer', () => {
  it('should return a wrappedComponent', () => {
    let key;
    let props;
    let reducer;
    const context = {
      store: {
        injectReducer: jest.fn(key, reducer),
      },
    };
    const MockComponent = () => <div>Component</div>;
    const WrapperComponent = withReducer(key, reducer)(MockComponent);
    const wrapper = shallow(<WrapperComponent {...props} />, { context });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
