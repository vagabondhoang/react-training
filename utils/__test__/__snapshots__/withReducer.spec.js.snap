// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReducer should return a wrappedComponent 1`] = `<MockComponent />`;
